Readme.md

```diff
@@ -6,20 +6,18 @@
 
 # DevExpress Blazor Components - Set predefined settings for the specific component
 
-Controls usually have the same settings for different pages of a large web application, and we recommend that you specify the component settings in markup. You can also override the **SetParametersAsync** method to specify the default settings for Blazor components in your application. However, we do not recommend that you use this approach because it is not a common approach in Blazor. Use it at your own risk.
+Controls usually have the same settings for different pages of a large web application, and we recommend that you specify the component settings in markup. You can also override the **SetParametersAsyncCore** (for Grid) or **SetParametersAsync** (for other components) method to specify the default settings for Blazor components in your application. However, we do not recommend that you use this approach because it is not a common approach in Blazor. Use it at your own risk.
 
 ![Grid with predefined settings and default](images/result.png)
 
-Create an inheritor from our Blazor control that overrides the **SetParametersAsync** method. In this method, set the default values for the public properties before you call the base method. You can initialize properties with custom default values in markup because the base method is called after your custom settings. We do not recommend that you override and change the values of protected properties.
+Create an inheritor from our Blazor control that overrides the **SetParametersAsync** or **SetParametersAsyncCore** method. In this method, set the default values for the public properties before you call the base method. You can initialize properties with custom default values in markup because the base method is called after custom settings are applied. We do not recommend that you override and change the values of protected properties.

```

Comment:
> Create an inheritor from our Blazor control that overrides the **SetParametersAsync** or **SetParametersAsyncCore** method.

I see a logical flaw in this sentence. You (or, rather, the original author) clearly meant that "an inheritor" overrides the mentioned methods. However, it reads "our Blazor control that overrides the **SetParameterAsync** or **SetParameterAsyncCore** method."

> In this method, set the default values for the public properties before you call **the base method**.

I'm unsure what you meant by "the base method". Do you speak of the corresponding (overridden) method in the base class (that is, your Blazor component class).

> You can initialize properties with **custom default values**

I have only one question. How default values can be custom or vice versa? It sounds like an oxymoron to me.

> We do not recommend that you override and change the values of protected properties.

We could have clarified what would happen otherwise, that is, a rationale for this kind of limitation.

Reply:
> We could have clarified what would happen otherwise, that is, a rationale for this kind of limitation.

In this case, we cannot guarantee the correct functioning of our components. That is why these properties are not public but protected.

```suggestion
Inherit from our Blazor control to override the **SetParametersAsync** or **SetParametersAsyncCore** method. In this method, set the default values for the public properties before you call the base **SetParametersXXX** method. In markup, you can initialize properties with other than pre-defined default values because the base method is called after custom settings are applied. We do not recommend that you override and change the values of protected properties.

```

Reply:
I often say that one shouldn't rely on non-public APIs mainly because they are subject to change in future builds. We can only guarantee backward compatibility for the "white" or "official" APIs.

Regarding your change:

```suggestion
Inherit from our Blazor control to override the **SetParametersAsync** or **SetParametersAsyncCore** method. In one of these methods, set the default values for the public properties before you call the base **SetParametersXXX** method. In markup, you can initialize properties with modified default values because the base method is called after custom settings are applied. We do not recommend that you override and change the values of protected properties.
```